python/ray/tests/test_k8s_operator_examples.py

```diff
@@ -0,0 +1,157 @@
+"""Tests launch and teardown of multiple Ray clusters using Kubernetes
+operator."""
+import sys
+import os
+import subprocess
+import tempfile
+import time
+import unittest
+
+import kubernetes
+import pytest
+import yaml
+
+IMAGE_ENV = "KUBERNETES_OPERATOR_TEST_IMAGE"
+IMAGE = os.getenv(IMAGE_ENV, "rayproject/ray:nightly")
+NAMESPACE = "test-k8s-operator-examples"
+
+
+def retry_until_true(f):
+    def f_with_retries(*args, **kwargs):
+        while True:
+            if f(*args, **kwargs):
+                break
+            else:
+                time.sleep(1)
+
+    return f_with_retries
```

Comment:
should add a timeout to this. there's similar functionality already in `ray.test_utils.wait_for_condition` that you could use as well

Reply:
Changed to try once a second for thirty seconds.